src/control/server/storage/types.go

```diff
@@ -71,33 +71,62 @@ type (
 
 	// ScmNamespaces is a type alias for []ScmNamespace that provides a fmt.Stringer implementation.
 	ScmNamespaces []ScmNamespace
+
+	// ScanRequest defines the parameters for a Scan operation.
+	ScmScanRequest struct {
+		Rescan bool
+	}
+
+	// ScanResponse contains information gleaned during a successful Scan operation.
+	ScmScanResponse struct {
+		State      ScmState
+		Modules    ScmModules
+		Namespaces ScmNamespaces
+	}
```

Comment:
If I understand correctly, the reason these were moved up was to add the stringer and allow the ScanResponse struct to be passed in protobuf messages so that it can be used in the frontend code.

I don't agree with this change because it's a dilution of responsibilities. The goal of putting those request/response structs in the scm package was to keep them tightly focused on the responsibilities of making requests and conveying responses. By making them have more responsibility for displaying output to the user we lose that tight focus and introduce undesired coupling between different areas of the codebase.

Reply:
so you are suggesting that we duplicate the structures ?

Reply:
No, I'm suggesting that there's no reason for anything outside of the server package to know anything about making requests to and receiving responses from the SCM storage provider. We've already moved the modules/namespaces/states types into the storage package.

Reply:
To put it another way... Why should the dmg code know anything about the details of how the server performs a scan? More to the point, why should we have to make changes to dmg code because we changed some detail in the scm provider?

From the [SOLID go talk](https://dave.cheney.net/2016/08/20/solid-go-design):

> Why is it important that a piece of code should have only one reason for change? Well, as distressing as the idea that your own code may change, it is far more distressing to discover that code your code depends on is changing under your feet. And when your code does have to change, it should do so in response to a direct stimuli, it shouldn’t be a victim of collateral damage.
> 
> So code that has a single responsibility therefore has the fewest reasons to change.